### Investigations

The following are a few unexplained obervations from our project, and potential hypotheses to answer these anomalies.
1) Obscured selling price for some vehicles
2) Unrealistic high odometer readings
3) Low condition ratings for vehicles with extremely low mileage
4) Older vehicles with high selling prices
5) Multiple vehicles with odometer readings in the hundreds of thousands

### Hypotheses
1) Why were some vehicles selling for incredibly high prices
It is possible that some of these vehicles were high-performance luxury cars with high value

These vehicles were in pristine condition with low mileage

Could be an error 

2) Why did some vehicles have unrealistic high odometer readings?
This could be a data error 

Vehicle could have been some kind of project to test the engine's endurance

The vehicle was list with the incorrect information

3) Why did some vehicles have low condition ratings with low mileage?
Vehicle could have been involved in an accident

Could be an error in the data

The vehicle lost its value over time while sitting at a dealer and subsequently also worsened in condition

4) Why did some older vehicle sell with high price tags?
These vehicles could have been classics

The vehicles could have in incredibly nice condition and kept its value over time

This could have been another error 

5) Why do some vehicles have odomter readings in the hundreds of thousands?
Some vehicles are incredibly reliable and were built to last, this is not out of the ordinary

These vehicles were kept in good condition and had proper maintenance done, combined with engine reliability, they kept its value

Once again, this could be another error 

### Khalid 
## Unexplained observation #1

"Why were some vehicles selling for incredibly high prices"
This is not out of the ordinary, but it's worth to look into. Many olderm vehicles retain their value, especiialy if they are high-performance luxury sports cars.

We are going to plot what these vehicles prices were, condition, and mileage.



In [4]:
import pandas as pd

# Path to your CSV file
csv_path = '/Users/khali/Desktop/Coding Workspace/DataMining/data/cleaned/cleaned_data.csv'

# Load the CSV file into a DataFrame
df = pd.read_csv(csv_path)

# Check the first few rows to verify loading
print("Sample Data:")
print(df.head())

# Find the threshold for 'high selling price' (e.g., top 5% of prices)
high_price_threshold = df['sellingprice'].quantile(0.95)  # 95th percentile

# Filter rows where 'sellingprice' is greater than the threshold
high_price_vehicles = df[df['sellingprice'] > high_price_threshold]

# Select the relevant columns to display
columns_to_show = ['year', 'make', 'model', 'condition', 'odometer', 'sellingprice']
high_price_vehicles = high_price_vehicles[columns_to_show]

# Display the filtered DataFrame
print("\nVehicles with High Selling Price:")
print(high_price_vehicles)

# Save the result to a new CSV file (optional)
high_price_vehicles.to_csv('high_price_vehicles.csv', index=False)
print("\nFiltered data saved to 'high_price_vehicles.csv'")


Sample Data:
   year   make                model        trim   body transmission  \
0  2015    Kia              Sorento          LX    SUV    automatic   
1  2015    Kia              Sorento          LX    SUV    automatic   
2  2014    BMW             3 Series  328i SULEV  Sedan    automatic   
3  2015  Volvo                  S60          T5  Sedan    automatic   
4  2014    BMW  6 Series Gran Coupe        650i  Sedan    automatic   

                 vin state  condition  odometer  color interior  \
0  5xyktca69fg566472    ca        5.0   16639.0  white    black   
1  5xyktca69fg561319    ca        5.0    9393.0  white    beige   
2  wba3c1c51ek116351    ca       45.0    1331.0   gray    black   
3  yv1612tb4f1310987    ca       41.0   14282.0  white    black   
4  wba6b2c57ed129731    ca       43.0    2641.0   gray    black   

                                   seller      mmr  sellingprice  \
0                 kia motors america  inc  20500.0       21500.0   
1                 kia

### Investigation Concluded

After reviewing the CSV with pandas, we can come to the following conclusion: The higher priced vehicles were warrented due to their adequate attributes.
The 2014 BMW 6 Series Gran Coupe selling for 67,000 makes sense due its high condition rating at 43, as well as its incredibly low mileage of 2641.
Another example explained would be the 2015 Volvo S60 which sold for 27,750 but had reasonably low mileage at 14,282 as well as a high condition rating.
Basically, many of these high priced vehicles had price tags that made sense. The vehicle ended up being sold for what it was worth. 
In a previous assignment, we also investiagted other German cars that had similar coniditon ratings and mileage and still maintained their value and were sold for a lot.
One notable example was the 2014 BMW M5 which sold for 65,000. Considering it was an M series car, the unusual high price tag for a 2014 car made sense.
Further analysis will prove that unless there are errors in individual vehicle data, a lot of these cars are worth for what they are being sold for. 
